packages/react-material-ui/src/components/Table/hooks/useTableRoot.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import React, { createContext, useContext } from 'react';
+import { HeaderProps, RowProps, TableQueryStateProps } from '../types';
+
+type TableContextProps = {
+  rows: RowProps[];
+  headers: HeaderProps[];
+  total: number;
+  isControlled: boolean;
+  pageCount: number;
+  tableQuery: TableQueryStateProps;
+  selected: RowProps[];
+  isSelected: (id: string) => boolean;
+  handleChangePage: (event: unknown, newPage: number) => void;
+  handleChangeRowsPerPage: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleRequestSort: (
```

Comment:
Maybe the name of this attribute should be more explicit. What does `handleRequestSort` mean? If it's sorting by a request response, wouldn't `handleSortByRequestData` be a better name?

Reply:
This comes from the official material docs. Though in this case it's not related to a network request so I agree with you.

rename to `handleSort`.